File: backend/Models/OrderItem.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace BonelessPharmacyBackend
{
    /// <summary>
    /// A Order of items in the BonelessPharmacy system
    /// </summary>
    public class OrderItem
    {
        [Key]
        /// <summary>
        /// The primary key of the OrderItem
        /// </summary>
        /// <returns></returns>
        public int Id { get; set; }

        //TODO: Add properties for OrderId and SupplierCode
       
        /// <summary>
        /// The quantity of the Order
        /// </summary>
        /// <returns></returns>
        public int Quantity { get; set; }

        [Required]
        /// <summary>
        /// The price for the order, GST excluded
        /// </summary>
        /// <example>
        /// 420.60
        /// </example>
        /// <returns></returns>
        public double Price { get; set; }
    }
}

File: frontend/app/ts/Models/OrderItem.ts
interface OrderItem
{
    /**
    * The primary key of the Order Item
    */
    id: number;
    
    /**
     * The id of the Order item Order
     * orderId: number;
     */

    /**
     * The related Order object
     * order: Order;
     */

     /**
     * The id of the Order item Supplier
     * supplierCodeID: number;
     */

    /**
     * The related Supplier object
     * supplier: Supplier;
     */
    
    /**
    * Quantity of order
    */
    quantity: number;
    
    /**
    * The price of order
    */
    price: number;
}